lib/elixir/src/elixir_erl_for.erl

```diff
@@ -92,37 +94,51 @@ collect_filters([H | T], Acc) ->
 collect_filters([], Acc) ->
   {Acc, []}.
 
-build_inline(Ann, Clauses, Expr, Into, _Var, Acc, S) ->
+
+build_inline(Ann, Clauses, Expr, Into, true, _Var, Acc, S) ->
+  build_inline(Ann, Clauses, Expr, Into, true, Acc, S);
+build_inline(Ann, Clauses, Expr, Into, false, _Var, Acc, S) ->
   case lists:all(fun(Clause) -> element(1, Clause) == bin end, Clauses) of
```

Comment:
What about:

```erlang
build_inline(Ann, Clauses, Expr, Into, Uniq, _Var, Acc, S) ->
    case not Uniq and lists:all(fun(Clause) -> element(1, Clause) == bin end, Clauses) of
```

?

Both are fine, your choice. :)

Reply:
I started with this actually, because it is clearer, but I moved to the function clause because if I understood correctly the compiler does some optimizations moving those to the guard clause. I guess wasn't really necessary